nullaway/src/main/java/com/uber/nullaway/handlers/StreamNullabilityPropagator.java

```diff
@@ -229,10 +253,14 @@ public void onMatchMethodInvocation(
             observableCallToInnerMethodOrLambda.put(tree, argTree);
           }
         } else {
-          CollectLikeMethodRecord collectlikeMethodRecord =
-              streamType.getCollectlikeMethodRecord(methodSymbol);
-          if (collectlikeMethodRecord != null && methodSymbol.getParameters().length() == 1) {
-            handleCollectCall(tree, collectlikeMethodRecord);
+          if (methodSymbol.getParameters().length() == 1) {
+            for (CollectLikeMethodRecord collectlikeMethodRecord :
+                streamType.getCollectlikeMethodRecords(methodSymbol)) {
+              boolean handled = handleCollectCall(tree, collectlikeMethodRecord);
+              if (handled) {
+                break;
+              }
```

Comment:
This break is here because `collectlikeMethodRecords` is a multi-map, since each record has the outer `.collect(...)` method, plus info about the method call passed as it's argument, correct? This there can be multiple records for the same outer method call (`collectMethodSig`), but with different `collectorFactoryMethodClass` and `collectorFactoryMethodSig`.  `handleCollectCall(...)` has the full logic to test the `collectlikeMethodRecord` against the current invocation and, for a given invocation, only one record should ever match (though this isn't statically enforced).

We could potentially make this more robust by introducing a record with info on all the arguments our `StreamTypeRecord` supports for a given `collectMethodSig`. Then we can have the map be a single map to that compound record, which we can pass to `handleCollectCall(...)`, which can further query for the right `CollectLikeMethodRecord` within it (possibly by passing it `collectorFactoryMethodClass` and `collectorFactoryMethodSig`, which the outer record can use another map to look for).

We wouldn't even have to change the interface into `StreamModelBuilder`, just what the builder does underneath, and it could then easily check for duplicates or inconsistencies.

That said, I don't mind the current approach all that much either. Would appreciate a quick comment on why we are stopping iteration early here (i.e. why only one record should match), though.

Reply:
I see what you're saying here, and I agree it would be statically safer to map the `collectMethodSig` to another map keyed on the factory method info.  But, given how little code we have that is constructing `StreamTypeRecord`s, I'm not sure the further refactoring is worth it at this point.  I will add some comments on the code above to clarify what is going on.